packages/api-server/src/__tests__/withWebServer.test.ts

```diff
@@ -1,142 +1,281 @@
+import fs from 'fs'
 import path from 'path'
 
-import type { FastifyInstance, FastifyPluginCallback } from 'fastify'
+import { getPaths } from '@redwoodjs/project-config'
 
-import { loadFastifyConfig } from '../fastify'
+import { createFastifyInstance } from '../fastify'
 import withWebServer from '../plugins/withWebServer'
 
-const FIXTURE_PATH = path.resolve(
-  __dirname,
-  '../../../../__fixtures__/example-todo-main'
-)
-
-// Mock the dist folder from fixtures,
-// because its gitignored
-jest.mock('../plugins/findPrerenderedHtml', () => {
-  return {
-    findPrerenderedHtml: () => {
-      return ['about.html', 'mocked.html', 'posts/new.html', 'index.html']
-    },
-  }
-})
+// Suppress terminal logging.
+console.log = jest.fn()
 
-jest.mock('../fastify', () => {
-  return {
-    ...jest.requireActual('../fastify'),
-    loadFastifyConfig: jest.fn(),
-  }
-})
+// Set up RWJS_CWD.
+let original_RWJS_CWD
 
 beforeAll(() => {
-  process.env.RWJS_CWD = FIXTURE_PATH
+  original_RWJS_CWD = process.env.RWJS_CWD
+
+  process.env.RWJS_CWD = path.join(__dirname, 'fixtures/redwood-app')
 })
+
 afterAll(() => {
-  delete process.env.RWJS_CWD
+  process.env.RWJS_CWD = original_RWJS_CWD
 })
 
-test('Attach handlers for prerendered files', async () => {
-  const mockedFastifyInstance = {
-    register: jest.fn(),
-    get: jest.fn(),
-    setNotFoundHandler: jest.fn(),
-    log: console,
-  } as unknown as FastifyInstance
-
-  await withWebServer(mockedFastifyInstance, { port: 3000 })
-
-  expect(mockedFastifyInstance.get).toHaveBeenCalledWith(
-    '/about',
-    expect.anything()
-  )
-  expect(mockedFastifyInstance.get).toHaveBeenCalledWith(
-    '/mocked',
-    expect.anything()
-  )
-  expect(mockedFastifyInstance.get).toHaveBeenCalledWith(
-    '/posts/new',
-    expect.anything()
-  )
-
-  // Ignore index.html
-  expect(mockedFastifyInstance.get).not.toHaveBeenCalledWith(
-    '/index',
-    expect.anything()
-  )
-})
+// Set up and teardown the Fastify instance for each test.
```

Comment:
You say 'each test' here but use the 'beforeAll' and 'afterAll' jest hooks. I was worried I was not remembering how jest worked so I had a read through this [page on their docs](https://jestjs.io/docs/setup-teardown#scoping). I then tested it by adding a `console.error('beforeAll')` and it was called only once when running this test file. 

Just want to confirm this is the behaviour you wanted? This applies to the other test files too.

Reply:
Yeah that's the behavior I wanted, let me amend the comment

Reply:
The comment served more just to demarcate a section of the code, so just lopped off the end of it: https://github.com/redwoodjs/redwood/pull/9325/commits/ebd94a2c93030d435cd40eb12d782285f2f70d47